packages/auth-sdk/user.tsx

```diff
@@ -43,6 +43,10 @@ async function fetchAuthenticatedUser(apiOrigin: string): Promise<User | null> {
          typeof payload.unique_username === 'string',
          'User handle is not a string'
       );
+      invariant(
+         payload.unique_username !== null,
+         'User handle is null'
+      );
```

Comment:
The issue described by #372 is that if the user `unique_username` is not defined, then this hook does not return the user because of [this invariant](https://github.com/iFixit/react-commerce/blob/main/packages/auth-sdk/user.tsx#L42-L45).

In order for us to be able to support possibly null/undefined `unique_username` we need to get rid of the invariant, type the `unique_username` as nullable, and refactor the UI components that access `unique_username` unsafely, so that they take into account that the value might be null.

Reply:
Thanks! I wasn't sure about this and it makes sense now.

Reply:
Resolved via 19f4cc0085a3d47b323aacd4fd4cebfe183ac2d9, 16e184466f83edd6b9c09db6017b65258d2f95fd and dc101d2f422dc4106594c34ec45fffb03a2ea8e0